src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ILoggerExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal static class ILoggerExtensions
+    {
+        public static bool TestOnlyLoggingEnabled = false;
+
+        public static void LogTestOnly(this ILogger logger, string message)
+        {
+            if (!TestOnlyLoggingEnabled)
```

Comment:
Personally I'd remove the negation here, put the logger.LogDebug in this if and let the non-test case just fall out, but that's just a personal preference, this is probably fine.

Reply:
This was my weak attempt at micro-optimization, but I think with Taylors suggestion of a conditional attribute I can make this look nicer.